_posts/2015-10-21-Onefootbot.md

```diff
@@ -0,0 +1,123 @@
+---
+layout: post
+title: Onefootbot - the Onefootball chat robot
+permalink: /onefootbot-chat-robot
+tags: hubot, chatops
+author_id: dpahl
+category: devops
+---
+
+We've recently heard a lot about [ChatOps](https://www.pagerduty.com/blog/what-is-chatops/) and are thrilled by the 
+possibilities and chances. So we introduced Onefootbot as our company robot based on [Hubot](https://hubot.github.com/) 
+connected via [HipChat](https://www.hipchat.com/). [github](https://github.com/) wrote [Hubot](https://hubot.github.com/) 
+to automate their company chat room. Later they released it as open source. It is written in CoffeeScript on Node.js and 
+can easily be extended or customized by your own scripts. The general idea behind it is to bring your tools into your 
+conversations and use the chat bot to work with plugins and scripts so teams can automate tasks and collaborate, working 
+better, cheaper and faster.
+
+## What can Hubot do ?
+
+Hubot is shipped with a small group of core scripts in order to do things like posting images, translating languages and 
+integrating with Google Maps, etc. In order to extend these core functions you can choose from a number of 
+[Community packages](https://github.com/hubot-scripts) and configure them according to your needs. The real fun happens 
+when adding your own scripts. 
+
+Hubot is especially useful in connection with API's and web hooks when it comes to DevOps activities. Possible use cases are
+
+* Start a test run on the CI server
+* Trigger a build process
+* Deploy your app to staging and/or production
+
+## How to run your own robot ?
+
+In order to run your own robot you need to know a few basics. Hubot is following an event driven approach. In case of an 
+event to happen it executes a callback that is defined in your CoffeeScript source code. 
+
+An event where it reacts to can be a
+
+* message mentioning the robot in a chatroom or via a direct private message
+* text pattern that is detected in any message of a room where the robot is connected to
+* HTTP request from an external system
+
+A callback can do things like
+
+* reply to a message in the chatroom or in a private chat
+* return a HTTP response
+* trigger a new HTTP request
+* execute a shell command
+* whatever can be done with Node.js
+
+Please have a look into the Hubot [documentation](https://hubot.github.com/docs/scripting/) in order to get started with 
+your own hubot scripts.
+
+## So what can Onefootbot do?
```

Comment:
I thought it would be good like that so to explain what Hubot can do in general and what Onefootbot can do specific.